test/update.test.ts

```diff
@@ -101,53 +100,54 @@ describe.skip("update", () => {
     const now = Date.now()
     const update = await p.getUpdate(true)
 
-    assert.deepEqual({ update }, await settings.read())
-    assert.equal(isNaN(update.checked), false)
-    assert.equal(update.checked < Date.now() && update.checked >= now, true)
-    assert.equal(update.version, "4.1.1")
-    assert.deepEqual(spy, ["/latest"])
+    await expect(settings.read()).resolves.toEqual({ update })
+    expect(isNaN(update.checked)).toBe(false)
+    expect(update.checked < Date.now() && update.checked >= now).toBe(true)
+    expect(update.version).toBe("4.1.1")
+    expect(spy).toBe(["/latest"])
   })
 
   it("should get latest after interval passes", async () => {
     const p = provider()
     await p.getUpdate()
-    assert.deepEqual(spy, [])
+    expect(spy).toEqual([])
 
     let checked = Date.now() - 1000 * 60 * 60 * 23
     await settings.write({ update: { checked, version } })
     await p.getUpdate()
-    assert.deepEqual(spy, [])
+    expect(spy).toEqual([])
 
     checked = Date.now() - 1000 * 60 * 60 * 25
     await settings.write({ update: { checked, version } })
 
     const update = await p.getUpdate()
-    assert.notEqual(update.checked, checked)
-    assert.deepEqual(spy, ["/latest"])
+    expect(update.checked).not.toBe(checked)
+    expect(spy).toBe(["/latest"])
   })
 
   it("should check if it's the current version", async () => {
     version = "9999999.99999.9999"
 
     const p = provider()
     let update = await p.getUpdate(true)
-    assert.equal(p.isLatestVersion(update), false)
+    expect(p.isLatestVersion(update)).toBe(false)
 
     version = "0.0.0"
     update = await p.getUpdate(true)
-    assert.equal(p.isLatestVersion(update), true)
+    expect(p.isLatestVersion(update)).toBe(true)
 
     // Old version format; make sure it doesn't report as being later.
     version = "999999.9999-invalid999.99.9"
     update = await p.getUpdate(true)
-    assert.equal(p.isLatestVersion(update), true)
+    expect(p.isLatestVersion(update)).toBe(true)
   })
 
   it("should not reject if unable to fetch", async () => {
+    expect.assertions(2)
     let provider = new UpdateProvider("invalid", settings)
-    await assert.doesNotReject(() => provider.getUpdate(true))
+    await expect(() => provider.getUpdate(true)).resolves.toBe(undefined)
```

Comment:
Does Jest automatically handle if this is a function or does it need to be `expect(provider.getUpdate(true))`?

Reply:
Good question. Let me take a closer look!

Reply:
Two things:
1. this whole test suite is `skip`ed - not sure when that was, but I will eventually look into adding it back in.
2. Because `provider.getUpdate` returns a promise, we have to pass `expect` a function than assert with `.resolves` so this is correct

source: https://jestjs.io/docs/en/tutorial-async#asyncawait

Reply:
I thought the same thing from a long time ago when I used Jest once but in that link and in their docs they have examples like:

```
 await expect(user.getUserName(5)).resolves.toEqual('Paul');
```

Which doesn't pass a function.

We have several promise-based `expect` calls in other tests that don't use a function as well. For example in the cli tests we have:

```
    await expect(shouldOpenInExistingInstance(args)).resolves.toStrictEqual("test")
```

Maybe both methods work? Or one of them is wrong lol, idk

Reply:
OH! I see what you mean. I wonder if I did that because the promise resolves to undefined which throws an error? Maybe not! I'll write a note down to go through and look at that again. Sorry I didn't understand what you meant at first. Good eye!